### ISSUE TITLE
Figure out how to make Sentry public

### ISSUE THREAD

**Issue by calmh** [syncthing/syncthing#5784]
Ideally so links can be followed to get a read only view of the issue, with public signup to get an account with whatever the normal read/write-but-not-admin permissions entail.

**Comment by AudriusButkevicius** [comment]
```
Version 5.4.0
-------------

SENTRY_PUBLIC now dictates whether or not a Sentry install should be considered
accessible by all users or not. This should better solve the use-case of companies
hosting a Sentry instance internally and not necessarily needing the permissions
that teams give you.

If enabled, all teams and all projects will be accessible all members (for any endpoint
which does not require a certain level of access). Project.is_public now dictates the
implied state of all events, but will not allow a non-member to browse events.

...

Version 5.3.0
-------------

- Sentry now requires authentication for all pages.
- SENTRY_PUBLIC behavior has changed to signify the default state of projects.
```

So it seems registration is required which is sad.
Is it worth the effort? Who do we expect to read this?

**Comment by calmh** [comment]
I think requiring registration is fine, it’s a niche thing. Kinda like the build server. But is there a way to open registration? Currently it’s invitation only i think?

**Comment by calmh** [comment]
Looks like `SENTRY_FEATURES['auth:register']` is what we need and that's all. The project is already public. I don't expect very many people to be interested, but given that collecting data is already contentious I'd like to avoid the perception that we're keeping secret behind closed doors.

**Comment by AudriusButkevicius** [comment]
I guess do we plan to have anything private?

**Comment by calmh** [comment]
Probably not? FWIW the option above is documented to be the default, yet I've set it explicitly and there is still no signup. :( troubleshooting.

(and I futzed around with the docker config to ensure it reads my sentry.conf.py; other changes take so that part seems fine.)

**Comment by AudriusButkevicius** [comment]
Changes to sentry.conf requires docker-compose build

**Comment by calmh** [comment]
Yes, as noted above I've done the needful.

**Comment by AudriusButkevicius** [comment]
We can set SENTRY_PUBLIC (which makes all projects public) and try to add github auth I think, which might remove the need to register (in a way).

**Comment by calmh** [comment]
Yeah that's an alternative. I've set PUBLIC. I'll try to add GitHub SSO and see what happens.

**Comment by calmh** [comment]
No effect. I appear defeated.

**Comment by calmh** [comment]
I think it *should* work because

```
root@sentry:/home/ab/onpremise# docker exec -it onpremise_web_1 tail /etc/sentry/sentry.conf.py

# Syncthing customization

# Enables users to sign up themselves
SENTRY_FEATURES['auth:register'] = True
# Enables the Custom Inbound Filters feature
SENTRY_FEATURES['projects:custom-inbound-filters'] = True
# All data is publicly accessible
SENTRY_PUBLIC = True

root@sentry:/home/ab/onpremise# docker exec -it onpremise_web_1 tail sentry.conf.py

# Syncthing customization

# Enables users to sign up themselves
SENTRY_FEATURES['auth:register'] = True
# Enables the Custom Inbound Filters feature
SENTRY_FEATURES['projects:custom-inbound-filters'] = True
# All data is publicly accessible
SENTRY_PUBLIC = True

root@sentry:/home/ab/onpremise#
```

and the `custom-inbound-filters` filters setting does work, enabling something that wasn't previously enabled in the UI...

**Comment by AudriusButkevicius** [comment]
Check the yaml files and compose files in case it's disabled and overriden there?
Also, obvious question, you did try restarting it right?

**Comment by calmh** [comment]
Yeah, checking now... Given the sentry.conf.py gets baked into the image, yes. `docker-compose down; docker-compose build; docker-compose up` and the `down` step even does `rm` on the old images, so this seems clean...

**Comment by calmh** [comment]
I added a `docker-compose run --rm web upgrade` after `build` for good measure, since this runs some sort of migration. No difference. :( You can take a look when you're home maybe, I'll leave this fight alone for now.

**Comment by AudriusButkevicius** [comment]
So I got registration to work, public =true is set as I've checked in the django shell, but it seems the user does not get assigned to any org after registration, hence can't see anything.

**Comment by calmh** [comment]
Funny thing is, googling this and looking around people seem to have mostly the opposite problem. :(
